workflow/controller/operator.go

```diff
@@ -1464,12 +1465,17 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 			woc.addChildNode(retryNodeName, nodeName)
 			node = nil
 
+			localParams := make(map[string]string)
 			// Change the `pod.name` variable to the new retry node name
 			if processedTmpl.IsPodType() {
-				processedTmpl, err = common.SubstituteParams(processedTmpl, map[string]string{}, map[string]string{common.LocalVarPodName: woc.wf.NodeID(nodeName)})
-				if err != nil {
-					return woc.initializeNodeOrMarkError(node, nodeName, wfv1.NodeTypeSkipped, templateScope, orgTmpl, opts.boundaryID, err), err
-				}
+				localParams[common.LocalVarPodName] = woc.wf.NodeID(nodeName)
+			}
+			localParams[common.LocalVarPodRetryAttempt] = strconv.Itoa(len(retryParentNode.Children))
```

Comment:
FYI I have reverted to original code but my rationale was to avoid the cost of doing `strconv.Itoa` (or the equivalent `fmt.Sprintf`) multiple times.

Reply:
Sure, but conventionally in Argo we do `Sprint` for int to string conversion.

```suggestion
			localParams[common.LocalVarPodRetryAttempt] = fmt.Sprint(len(retryParentNode.Children))
```

Reply:
Understood. no problem.